python/pydantic_core/core_schema.py

```diff
@@ -106,6 +106,7 @@ class CoreConfig(TypedDict, total=False):
     hide_input_in_errors: bool
     validation_error_cause: bool  # default: False
     coerce_numbers_to_str: bool  # default: False
+    regex_engine: Literal['rust-regex', 'python-re']  # default: 'rust-regex'
```

Comment:
In the interest of time today I have just pushed this missing item, I'd like to follow up with a proper test next week. 🙏